src/main/java/co/rsk/federate/signing/hsm/client/HSMClientProvider.java

```diff
@@ -49,33 +49,35 @@ public HSMClient getClient() throws HSMClientException {
         int version = this.hsmClientProtocol.getVersion();
         HSMClient client;
         logger.debug("[getClient] version: {}, keyId: {}", version, keyId);
-        switch (version) {
-            case 1:
-                client = new HSMClientVersion1(this.hsmClientProtocol);
+        if (version == 1) {
+            client = new HSMClientVersion1(this.hsmClientProtocol);
+        } else if (version == 2 || version == 3 || version == 4) {
```

Comment:
Like this, we'll end up with 10 conditions for the same thing

Reply:
Resolved